src/engine/QueryPlanner.cpp

```diff
@@ -1368,11 +1369,13 @@ void QueryPlanner::applyFiltersIfPossible(
   // in one go. Changing `row` inside the loop would invalidate the iterators.
   std::vector<SubtreePlan> addedPlans;
   for (auto& plan : row) {
-    if (plan._qet->getType() == QueryExecutionTree::SCAN &&
-        plan._qet->getResultWidth() == 3 && !replace) {
-      // Do not apply filters to dummies, except at the very end of query
-      // planning.
-      continue;
+    if constexpr (!replace) {
```

Comment:
Is this `if constexpr` there for compilation reasons, or only for efficiency?
I don't think that the short-circuiting runtime if would be so expensive, but I have no strong option here again.